include/mvdtool/mvd_generic.hpp

```diff
@@ -57,30 +56,31 @@ inline MVDFile* open_mvd(const std::string & filename) {
 /// performed, but no guarantees about file correctness are made.
 ///
 /// \param filename the path of the file to open
+/// \param population population parameter
 /// \return a shared pointer to a mvd::File object
 ///
 inline std::shared_ptr<File> open(const std::string& filename,
                                   const std::string& population="") {
     std::shared_ptr<File> mvdfile;
-    switch(_mvd_format(filename)) {
-        case MVDType::MVD2:
-            throw std::runtime_error("MVD2 not supported in the new MVD API");
-        case MVDType::MVD3:
-            try {
-                mvdfile.reset(new MVD3::MVD3File(filename));
-                mvdfile->size();
-            } catch (...) {
-                throw std::runtime_error("invalid MVD3 file");
-            }
-            break;
-        default:
-            try {
-                mvdfile.reset(new SonataFile(filename, population));
-                mvdfile->size();
-            } catch (...) {
-                throw std::runtime_error("invalid Sonata file");
-            }
-            break;
+    switch (_mvd_format(filename)) {
+    case MVDType::MVD2:
+        throw std::runtime_error("MVD2 not supported in the new MVD API");
+    case MVDType::MVD3:
+        try {
+            mvdfile.reset(new MVD3::MVD3File(filename));
+            mvdfile->size();
```

Comment:
what does this do? Judging from the name of the method this statement doesn't do much?

Reply:
It should have a comment. Basically we do a quick operation to ensure the open was successful.